touchdown/ssh/client.py

```diff
@@ -52,16 +72,20 @@ def _run(self, transport, command):
             exit_status_ready = channel.exit_status_ready()
             while not exit_status_ready:
                 while channel.recv_ready():
-                    print(channel.recv(1024))
+                    buf = self._maybe_decode(channel.recv, 1024, input_encoding)
+                    print(buf, file=stdout)
```

Comment:
So to keep get_input_encoding working and use the existing logging stuff i guess we'll need a stub stdout wrapper that calls plan.echo() and that will need passing in in touchdown.aws.ec2.ami and touchdown.provisioner.provisioner. But that can wait for now if you get the tests to pass on 2.7.